dgv/metrics/task_functions.py

```diff
@@ -126,329 +238,296 @@ def get_dict(df, obj_property):
+def get_unique_dates(
+    first_list: list[str], second_list: Optional[list[str]]
+) -> list[str]:
+    """
+    Returns a list of unique dates from two input lists.
 
+    Args:
+        first_list (list[str]): The primary list of dates as strings.
+        second_list (Optional[list[str]]): The secondary list of dates as strings. Can be None.
 
-def get_unique_dates(first_list, second_list):
+    Returns:
+        list[str]: The list of unique dates.
+    """
     in_first = set(first_list)
-    in_second = set(second_list)
+    in_second = set(second_list) if second_list else set()
     in_second_but_not_in_first = in_second - in_first
-    result = first_list + list(in_second_but_not_in_first)
+    result = list(in_first) + list(in_second_but_not_in_first)
     return result
 
 
-def process_log(ti):
-    new_logs = ti.xcom_pull(key="new_logs", task_ids="get_new_logs")
-    newlogs = [nl.replace("/new/", "/ongoing/") for nl in new_logs]
-    all_dates_processed = []
-    print("downloading files...")
-    for nl in newlogs:
-        minio_infra.download_files(
-            list_files=[
-                {
-                    "source_path": "metrics-logs/ongoing/",
-                    "source_name": nl.split("/")[-1],
-                    "dest_path": TMP_FOLDER,
-                    "dest_name": nl.split("/")[-1]
-                }
+def aggregate_obj_type(log_date: str, obj_config: DataGouvObject) -> list[str]:
+    """
+    Calculate aggregated usage metrics by date and resource type.
+
+    Args:
+        log_date (str): aggregation date.
+        obj_config (DataGouvObject): object to aggregate (eg. dataset, resource..).
+
+    Returns:
+        list[str]: List of the processed dates.
+    """
+    try:
+        logging.info(f"---- {obj_config.type} ----")
+        df_catalog = pd.read_csv(
+            f"{obj_config.catalog_destination_path}{obj_config.catalog_destination_name}",
+            dtype=str,
+            sep=";",
+            usecols=obj_config.get_catalog_columns(),
+        )
+        df = pd.read_csv(
+            f"{TMP_FOLDER}found/found_{obj_config.type}.csv",
+            dtype=str,
+            sep=";",
+            header=None,
+        )
+        if obj_config.type not in ["resources"]:
+            catalog_dict = get_dict(df_catalog, "slug")
+            df["id"] = df[1].apply(
+                lambda x: catalog_dict[x] if x in catalog_dict else None
+            )
+            df = df.rename(columns={0: "date_metric", 2: "segment"})
+            df = df.drop(columns=[1])
+        else:
+            df = df.rename(columns={0: "date_metric", 1: "id", 2: "segment"})
+            df = pd.merge(df, df_catalog[["id", "url"]], on="id", how="left")
+            df["is_static"] = df["url"].apply(
+                lambda x: True if "static.data.gouv.fr" in str(x) else False
+            )
+            logging.info("shape with static", df.shape[0])
+            df = df[not df["is_static"], ["date_metric", "id", "segment"]]
```

Comment:
I am not sure whether `not df["is_static"]` works with the production version of `pandas`, could be safer to use:
```suggestion
            df = df[~df["is_static"], ["date_metric", "id", "segment"]]
```

Reply:
Fixing it 😃